File: Complementar - Estruturas de dados (Didático)/Algoritmos05/Alg05.py
'''
♦ Algoritmo 05: Estrutura de Dados Linear Estática (não Dinâmica) de Classificação/Ordenação Insertion Sort (Ordenação pelo Método (Iterativo) de Inserção).
'''

def insertionSort(array):
    n = len(array)
    
    for i in range(1, n):
        chave = array[i]
        j = i - 1
        
        while j >= 0 and array[j] > chave:
            array[j + 1] = array[j]
            j -= 1
        
        array[j + 1] = chave

array = [64, 34, 25, 12, 22, 11, 90]

insertionSort(array)

print("Array ordenado:")
for elemento in array:
    print(elemento, end=" ")

File: Complementar - Estruturas de dados (Didático)/Algoritmos05/Alg05.cpp
/* 
♦ Algoritmo 05: Estrutura de Dados Linear Estática (não Dinâmica) de Classificação/Ordenação Insertion Sort (Ordenação pelo Método (Iterativo) de Inserção).
*/

#include <iostream>
using namespace std;

void insertionSort(int array[], int n) {
    for (int i = 1; i < n; i++) {
        int chave = array[i];
        int j = i - 1;
        
        while (j >= 0 && array[j] > chave) {
            array[j + 1] = array[j];
            j--;
        }
        
        array[j + 1] = chave;
    }
}

int main() {
    int array[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(array) / sizeof(array[0]);
    
    insertionSort(array, n);
    
    cout << "Array ordenado:" << endl;
    for (int i = 0; i < n; i++) {
        cout << array[i] << " ";
    }
    
    return 0;
}
